src/Microsoft.Azure.ServiceBus/QueueClient.cs

```diff
@@ -284,6 +284,11 @@ internal SessionPumpHost SessionPumpHost
                 await this.innerReceiver.CloseAsync().ConfigureAwait(false);
             }
 
+            if (this.sessionClient != null)
+            {
+                await this.sessionClient.CloseAsync().ConfigureAwait(false);
+            }
+
             this.sessionPumpHost?.Close();
```

Comment:
>this.sessionPumpHost?.Close(); [](start = 12, length = 30)

sessionPumpHost uses the sessionClient. So can you please close the pumphost first, then close the sessionclient..